### PR TITLE
Update Chinese translation of fornt-end-libraries

### LINKED ISSUE #40481: Chinese challenges (Front End Libraries) needing an audit for English changes from Dec, 1, 2019
@freeCodeCamp/chinese 

In preparation of the upcoming launch of the Chinese curriculum, the following Chinese curriculum challenges need to be compared to their English equivalents.

This issue only covers challenges in the **`Front End Libraries`** certification.

All the challenges below have Description or Instructions sections with changes to the English version since Dec 1, 2019. Also includes any challenges with Tests where at least one test's text has changed in the English version since Dec 1, 2019 or where the number of tests in the current English version is different than the number of tests in the current Chinese version.

Please create any necessary PRs to bring the Chinese challenge listed below in sync with the English version.  

**NOTES:**
- Some changes are not translation related (like addition of code examples or addition/removal of code tags (single back ticks).  These changes are still needed to bring make the challenge more like the English version.
- There are two links to the right of the challenge title (one allows you directly link to the editable Chinese version of the file on GitHub and the other links to the raw English version.  You can right-click and open the links in new tabs to keep this issue in the background.
---
- [ ] **Build a Random Quote Machine** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/front-end-libraries-projects/build-a-random-quote-machine.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-random-quote-machine.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**

</details>

---
- [ ] **Build a Drum Machine** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/front-end-libraries-projects/build-a-drum-machine.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-drum-machine.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**

</details>

---
- [ ] **Build a Markdown Previewer** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/front-end-libraries-projects/build-a-markdown-previewer.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-markdown-previewer.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**

</details>

---
- [ ] **Build a JavaScript Calculator** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/front-end-libraries-projects/build-a-javascript-calculator.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-javascript-calculator.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**

</details>

---
- [ ] **Use Comments to Clarify Code** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/bootstrap/use-comments-to-clarify-code.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/bootstrap/use-comments-to-clarify-code.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**
- **The following test(s) `text` do not match:**
**Instructions:** Make sure to also check that the English tests have not been reorderd.
**Test 2**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your comment should have the text <code>Only change code above this line</code>.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your comment should have the text `Code below this line should not be changed`.

```
  </td>
  </tr>
  </table>

</details>

---
- [ ] **Connect Redux to the Messages App** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react-and-redux/connect-redux-to-the-messages-app.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react-and-redux/connect-redux-to-the-messages-app.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **The following test(s) `text` do not match:**
**Instructions:** Make sure to also check that the English tests have not been reorderd.
**Test 2**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  The <code>Presentational</code> component should render an <code>h2</code>, <code>input</code>, <code>button</code>, and <code>ul</code> elements.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  The `Presentational` component should render to page.

```
  </td>
  </tr>
  </table>

</details>

---
- [ ] **Extract Local State into Redux** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react-and-redux/extract-local-state-into-redux.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react-and-redux/extract-local-state-into-redux.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **The following test(s) `text` do not match:**
**Instructions:** Make sure to also check that the English tests have not been reorderd.
**Test 2**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  The <code>Presentational</code> component should render an <code>h2</code>, <code>input</code>, <code>button</code>, and <code>ul</code> elements.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  The `Presentational` component should render to page.

```
  </td>
  </tr>
  </table>

</details>

---
- [ ] **Store Data with Sass Variables** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/sass/store-data-with-sass-variables.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/sass/store-data-with-sass-variables.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **The following test(s) `text` do not match:**
**Instructions:** Make sure to also check that the English tests have not been reorderd.
**Test 3**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your <code>.blog-post</code> element should have a </code>color</code> of red.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your `.blog-post` element should have a `color` of red.

```
  </td>
  </tr>
  </table>

**Test 4**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your <code>h2</code> elements should have a </code>color</code> of red.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your `h2` elements should have a `color` of red.

```
  </td>
  </tr>
  </table>

</details>

---
- [ ] **Create Reusable CSS with Mixins** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/sass/create-reusable-css-with-mixins.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/sass/create-reusable-css-with-mixins.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**
- **The following test(s) `text` do not match:**
**Instructions:** Make sure to also check that the English tests have not been reorderd.
**Test 2**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your code should include the <code>-webkit-border-radius</code> vender prefix that uses the <code>$radius</code> parameter.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your code should include the `-webkit-border-radius` vendor prefix that uses the `$radius` parameter.

```
  </td>
  </tr>
  </table>

**Test 3**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your code should include the <code>-moz-border-radius</code> vender prefix that uses the <code>$radius</code> parameter.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your code should include the `-moz-border-radius` vendor prefix that uses the `$radius` parameter.

```
  </td>
  </tr>
  </table>

**Test 4**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your code should include the <code>-ms-border-radius</code> vender prefix that uses the <code>$radius</code> parameter.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your code should include the `-ms-border-radius` vendor prefix that uses the `$radius` parameter.

```
  </td>
  </tr>
  </table>

</details>

---
- [ ] **Use @for to Create a Sass Loop** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/sass/use-for-to-create-a-sass-loop.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/sass/use-for-to-create-a-sass-loop.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Instructions` section has changed**
- **The following test(s) `text` do not match:**
**Instructions:** Make sure to also check that the English tests have not been reorderd.
**Test 2**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your <code>.text-1</code> class should have a <code>font-size</code> of 10px.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your `.text-1` class should have a `font-size` of 15px.

```
  </td>
  </tr>
  </table>

**Test 3**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your <code>.text-2</code> class should have a <code>font-size</code> of 20px.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your `.text-2` class should have a `font-size` of 30px.

```
  </td>
  </tr>
  </table>

**Test 4**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your <code>.text-3</code> class should have a <code>font-size</code> of 30px.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your `.text-3` class should have a `font-size` of 45px.

```
  </td>
  </tr>
  </table>

**Test 5**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your <code>.text-4</code> class should have a <code>font-size</code> of 40px.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your `.text-4` class should have a `font-size` of 60px.

```
  </td>
  </tr>
  </table>

**Test 6**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  Your <code>.text-5</code> class should have a <code>font-size</code> of 50px.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  Your `.text-5` class should have a `font-size` of 75px.

```
  </td>
  </tr>
  </table>

</details>

---
- [ ] **Change Text Inside an Element Using jQuery** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/jquery/change-text-inside-an-element-using-jquery.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/jquery/change-text-inside-an-element-using-jquery.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**

</details>

---
- [ ] **Use a Ternary Expression for Conditional Rendering** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/use-a-ternary-expression-for-conditional-rendering.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/use-a-ternary-expression-for-conditional-rendering.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**

</details>

---
- [ ] **Compose React Components** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/compose-react-components.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/compose-react-components.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Instructions` section has changed**

</details>

---
- [ ] **Write a React Component from Scratch** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/write-a-react-component-from-scratch.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/write-a-react-component-from-scratch.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **Number of tests do not match - English: 4 | Chinese: 3**
**Instructions:** Add or delete the applicable tests.  It is possible some tests have been rearranged, so make sure each English test has a corresponding Chinese test in the same order

</details>

---
- [ ] **Use State to Toggle an Element** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/use-state-to-toggle-an-element.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/use-state-to-toggle-an-element.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**

</details>

---
- [ ] **Add Inline Styles in React** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/add-inline-styles-in-react.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/add-inline-styles-in-react.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Instructions` section has changed**

</details>

---
- [ ] **Introducing Inline Styles** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/introducing-inline-styles.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/introducing-inline-styles.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Description` section has changed**

</details>

---
- [ ] **Render Conditionally from Props** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/render-conditionally-from-props.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/render-conditionally-from-props.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Instructions` section has changed**

</details>

---
- [ ] **Use the Lifecycle Method componentDidMount** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/use-the-lifecycle-method-componentdidmount.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/use-the-lifecycle-method-componentdidmount.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Instructions` section has changed**

</details>

---
- [ ] **Pass State as Props to Child Components** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/pass-state-as-props-to-child-components.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/pass-state-as-props-to-child-components.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **`Instructions` section has changed**

</details>

---
- [ ] **Use PropTypes to Define the Props You Expect** [Edit Chinese](https://github.com/freeCodeCamp/freeCodeCamp/edit/master/curriculum/challenges/chinese/03-front-end-libraries/react/use-proptypes-to-define-the-props-you-expect.md) | [View English](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/03-front-end-libraries/react/use-proptypes-to-define-the-props-you-expect.md)

<details><summary>Show/Hide sections with changes/issues</summary>

- **The following test(s) `text` do not match:**
**Instructions:** Make sure to also check that the English tests have not been reorderd.
**Test 3**
    <table>
  <tr>
  <th>
  Version
  </th>
  <th align="left">
  Text
  </th>
  </tr>
  <tr>
  <td align>Old</td>
  <td align>

```
  The <code>Items</code> component should include a <code>propTypes</code> check that requires <code>quantity</code> to be a <code>number</code>.
```
  </td>
  </tr>
  <td>New</td>
  <td>

```
  The `Items` component should include a `propTypes` check to require a value for `quantity` and ensure that its value is a number.

```
  </td>
  </tr>
  </table>

</details>

### PR DESCRIPTION
**Pull request by ZhichengChen**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40481

### CODE REVIEW

**Comment by ZhichengChen** [comment]
@miyaliu666 @S1ngS1ng Please review this PR.

**Review by miyaliu666 — DISMISSED** [review]
We'll update some descriptions on Crowdin, like constituting a more objective description for the user stories.
Thanks for your contribution! @ZhichengChen

File: curriculum/challenges/chinese/03-front-end-libraries/sass/use-for-to-create-a-sass-loop.md [hunk 1]
```diff
@@ -9,9 +9,9 @@ forumTopicId: 301462
 
 你可以在 Sass 中使用`@for`循环，它的表现类似与 JavaScript 中的`for`循环。
 
-`@for`以两种方式使用："start through end" 或 "start to end"。主要区别在于“开始结束”*排除* 结束号码，而“开始结束”*包括* 结束号码。
+`@for`以两种方式使用："开始 through 到" 或 "开始 to 结束"。主要区别在于“开始结束”*排除* 结束号码，而“开始结束”*包括* 结束号码。
 
-这是一个开始 **到** 结束示例：
+这是一个开始 **through** 结束示例：
```
Comment (RandellDawson) [inline curriculum/challenges/chinese/03-front-end-libraries/sass/use-for-to-create-a-sass-loop.md:14]: The word "through" should be translated.  It is not code.

Comment (RandellDawson) [inline curriculum/challenges/chinese/03-front-end-libraries/sass/use-for-to-create-a-sass-loop.md:14]:
```suggestion
这是一个开始 **through** 结束示例：
```

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
@ZhichengChen Thanks for your contribution.

**Pull request merged**